client/packages/invoices/src/Prescriptions/DetailView/SidePanel/PatientDetails.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { memo } from 'react';
+import {
+  Grid,
+  DetailPanelSection,
+  PanelLabel,
+  PanelRow,
+  useTranslation,
+  Autocomplete,
+} from '@openmsupply-client/common';
+import { usePrescription } from '../../api';
+import { useDiagnosisOptions } from '../../api/hooks/useDiagnosisOptions';
+
+export const PatientDetailsComponent = () => {
+  const t = useTranslation();
+
+  const {
+    query: { data },
+    isDisabled,
+    update: { update },
+  } = usePrescription();
+  const { diagnosis } = data ?? {};
+
+  const {
+    query: { data: diagnosisOptions },
+  } = useDiagnosisOptions();
+
+  return (
+    <DetailPanelSection title={t('heading.patient-details')}>
+      <Grid container gap={0.5}>
+        <>
+          <PanelRow style={{ marginTop: 12 }}>
+            <PanelLabel>{t('heading.diagnosis')}</PanelLabel>
+            <Autocomplete
+              fullWidth
+              clearable
+              options={diagnosisOptions ?? []}
+              value={{
+                label: diagnosis?.description ?? '',
+                value: diagnosis?.id ?? '',
+                id: diagnosis?.id ?? '',
+              }}
+              onChange={(_e, selected) => {
+                if (selected) {
+                  update({ diagnosisId: selected.value });
+                } else {
+                  // Updated value needs to be null for nullable input to work correctly
+                  update({ diagnosisId: null });
+                }
+              }}
+              disabled={isDisabled}
```

Comment:
You get a bit of an annoying flicker when selecting/changing the option. This is due to the value not actually being updated until the round-trip to/from the server is complete.

We could use the `useBufferState` hook, but that uses the ill-advised `useEffect` to update state, so we should really only use it when the external value can possibly be different to the locally selected value. So in this case, I'd just use this pattern, where keep a local value, but fall back to the external value when local state is not yet defined (i.e when first loading).

Here's a suggestion branch demonstrating: https://github.com/msupply-foundation/open-msupply/commit/e6e1143a152b6958626866b05a0cb1ad0c7d4be3

If you throttle your network speed significantly and compare the difference, it's quite noticeable.